core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -40,1372 +41,1024 @@ private CarbonCommonConstants() {
-   * value will avoid data screw problem for a large segment.
-   * Refer to CarbonData-1373 for more details.
+   * CARBON_TIMESTAMP
    */
-  @CarbonProperty
-  public static final String CARBON_UPDATE_SEGMENT_PARALLELISM =
-      "carbon.update.segment.parallelism";
+  public static final String CARBON_TIMESTAMP_MILLIS = "dd-MM-yyyy HH:mm:ss:SSS";
 
-  /**
-   * In default we will not optimize the update
-   */
-  public static final String CARBON_UPDATE_SEGMENT_PARALLELISM_DEFAULT = "1";
 
-  /**
-   * The property to configure the mdt file folder path, earlier it was pointing to the
-   * fixed carbon store path. This is needed in case of the federation setup when user removes
-   * the fixed store path nameService
-   */
-  @CarbonProperty
-  public static final String CARBON_UPDATE_SYNC_FOLDER = "carbon.update.sync.folder";
+  public static final Property CARBON_DATE_FORMAT = Property.buildStringProperty()
+      .key("carbon.date.format")
+      .defaultValue("yyyy-MM-dd")
+      .doc("Property for specifying the format of DATE data type column." +
+          " e.g. yyyy/MM/dd , or using default value")
+      .build();
 
-  public static final String CARBON_UPDATE_SYNC_FOLDER_DEFAULT = "/tmp/carbondata";
+  public static final Property COMPRESSOR = Property.buildStringProperty()
+      .key("carbon.column.compressor")
+      .defaultValue("snappy")
+      .doc("compressor for writing/reading CarbonData file")
+      .build();
 
-  /**
-   * Configures the parser/writer to limit the length of displayed contents being parsed/written
-   * in the exception message when an error occurs.
-   * Here {@code 0} means no exceptions will include the content being manipulated in their
-   * attributes.
-   */
-  public static final int CARBON_ERROR_CONTENT_LENGTH = 0;
+  public static final Property LOCK_TYPE = Property.buildStringProperty()
+      .key("carbon.lock.type")
+      .defaultValue("LOCALLOCK")
+      .doc("ZOOKEEPER_ENABLE_LOCK if this is set to true then zookeeper will" +
+          " be used to handle locking mechanism of carbon")
+      .build();
 
-  /**
-   * if the byte size of streaming segment reach this value,
-   * the system will create a new stream segment
-   */
-  @CarbonProperty
-  public static final String HANDOFF_SIZE = "carbon.streaming.segment.max.size";
+  public static final Property LOCK_PATH = Property.buildStringProperty()
+      .key("carbon.lock.path")
+      .defaultValue("")
+      .doc("Specifies the path where the lock files have to be created. " +
+          "By default, lock files are created in table path.")
+      .build();
 
   /**
-   * the default handoff size of streaming segment, the unit is byte
+   * ZOOKEEPER_LOCATION this is the location in zookeeper file system where locks are created.
+   * mechanism of carbon
    */
-  public static final long HANDOFF_SIZE_DEFAULT = 1024L * 1024 * 1024;
+  public static final String ZOOKEEPER_LOCATION = "/CarbonLocks";
```

Comment:
can you move this constaint to the end of the file

Reply:
ok, done